Library/Homebrew/cmd/deps.rb

```diff
@@ -80,9 +80,7 @@ def deps
       return
     end
 
-    installed = args.installed? || ARGV.formulae.all?(&:opt_or_installed_prefix_keg)
-
-    @use_runtime_dependencies = installed && recursive &&
+    @use_runtime_dependencies = args.installed? && recursive &&
```

Comment:
Sorry, this isn't desirable as it's considerably slower and the results aren't consistent with the output of other commands (e.g. `brew uninstall`). Instead you could try to move the `@use_runtime_dependencies` definition above `if args.tree?` which should mean this is used consistently.

Reply:
This change wasn't enough. I had to also make another in `recursive_deps_tree` function.